knoten/csm.py

```diff
@@ -41,6 +42,28 @@ def get_radii(camera):
     semi_minor = ellipsoid.getSemiMinorRadius()
     return semi_major, semi_minor
 
+def get_surface_normal(sensor, ground_pt):
+    """
+    Given a sensor model and ground point, calculate the surface normal.
+
+    Parameters
+    ----------
+    sensor : object
+             A CSM compliant sensor model object
+
+    ground_pt: tuple
+               The ground point as an (x, y, z) tuple
+
+    Returns
+    -------
+     : tuple
+       in the form (x, y, z)
+    """
+    semi_major, semi_minor = get_radii(sensor)
+
+    normal = spice.surfnm(semi_major, semi_major, semi_minor, np.array([ground_pt.x, ground_pt.y, ground_pt.z]))
+    return utils.Point(normal[0], normal[1], normal[2])
```

Comment:
This seems like something that can be converted to an object to be passed into functions as a shapemodel rather than depending on the sensor